core/src/kmx/kmx_processor.cpp

```diff
@@ -247,6 +247,7 @@ kmx_processor::internal_process_queued_actions(km_core_state *state) {
       }
       break;
     case QIT_INVALIDATECONTEXT:
+      state->context().clear();
```

Comment:
:+1:

But ... I worry that it may be possible to clear the context at the wrong time during the kmx processing? Is that possible?

Reply:
Great question I will do brief code inspection and see if It sounds correct.
internal_process_queued_actions is called 3 times

- [ x ] 1. kmx_processor::process_event *after* returning from _kmx.ProcessEvent. It is the final thing before returning

This should be absolutely fine it is not during the kmx processing.

- [ x ] 2. `kmx_processor::external_eventext`

This one is fine for now because the only event utilised for KM_CORE_EVENT_KEYBOARD_ACTIVATED. I think any future
external events that want do an action that causes the context to be cleared will be making that a deliberate action.

- [ ? ] 3. km_core_process_queued_actions->process_queued_actions->internal_process_queued_actions

This for the IMX keyboards...
Called in `kmhook_getmessage.cpp` line 285 
This case it can occur during kmx_processing of a keystroke, however the kmx_keyboard processors cache context will not be cleared I don't believe.

Reply:
Good analysis, thanks!